pandas/_typing.py

```diff
@@ -56,6 +56,7 @@
 FrameOrSeries = TypeVar("FrameOrSeries", bound="NDFrame")
 
 Axis = Union[str, int]
+Level = Union[str, int]
```

Comment:
```suggestion
Level = Optional[Hashable]
```

Reply:
technically this should be Optional since None is Hashable, but we don't currently support passing None as a parameter to level to indicate a level named None see #30848

so just Hashable may be fine. (would still need Optional[Level] in function signatures though)

Reply:
Nice catch. Took a slightly different approach for readability but I think covers the same concept